### PR TITLE
Create data parser

### PR DESCRIPTION
**Pull request by JonasBoecker**

### CODE REVIEW

File: core/dataprocessing/DataParser.hpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#include "string"
+
+namespace CAEMonitoringTool::DataProcessing {
+
+  using json = nlohmann::json;
+
+  class DataParser {
+
+  public:
+    static bool parseThreadInfo(const json &object,
```
Comment (julius-gummersbach) [inline core/dataprocessing/DataParser.hpp:16]: Documentation missing

File: core/datastore/DataManager.hpp [hunk 2]
```diff
@@ -3,82 +3,23 @@
...
-       */
-      void addThreadInfo(const json &threadInfo);
-
-    private:
 
-      /**
-       * general function pointer for the getter functions for the attributes of the threaddata objects
-       */
-      typedef const double &(*getterFunc)(const ThreadData &data);
+  class DataManager {
```
Comment (julius-gummersbach) [inline core/datastore/DataManager.hpp:14]: Documentation missing

**Review by julius-gummersbach — CHANGES_REQUESTED** [review]

**Review by sburk6 — APPROVED** [review]

**Review by TomsCodingCode — DISMISSED** [review]

File: core/dataprocessing/DataParser.hpp (hunk 1 above)
Comment (julius-gummersbach) [reply core/dataprocessing/DataParser.hpp:16]: ok later I guess

File: core/datastore/DataManager.hpp (hunk 2 above)
Comment (julius-gummersbach) [reply core/datastore/DataManager.hpp:14]: ok later I guess

**Review by sburk6 — DISMISSED** [review]

**Review by JosiasKasemann — APPROVED** [review]

**Review by sburk6 — APPROVED** [review]

**Review by julius-gummersbach — APPROVED** [review]

**Pull request merged**